stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/user/internal/DatabaseUserRepository.kt

```diff
@@ -32,11 +38,14 @@ internal class DatabaseUserRepository(
     // the user cache is simple, just keeps the last 100 users in memory
     private val userCache = LruCache<String, User>(cacheSize)
     private val latestUsersFlow: MutableStateFlow<Map<String, User>> = MutableStateFlow(emptyMap())
+    private val dbMutext = Mutex()
 
     override fun observeLatestUsers(): StateFlow<Map<String, User>> = latestUsersFlow
 
     override suspend fun clear() {
-        userDao.deleteAll()
+        scope.awaitWithMutex(dbMutext) {
```

Comment:
This isn't a real suspending function anymore now 🤔 . It now launches a separate coroutine and thus you can't rely on the clear() to have finished if you call it. Maybe in this case we should use:

    override suspend fun clear() {
        dbMutex.withLock {
            userDao.deleteAll()
        }
    }

Reply:
`scope.awwaitWithMutex(dbMutex) { }` is not launching a new coroutine, but blocking the current one

Reply:
Aha, nice :-).

Reply:
TBH, this extension function is not needed anymore, initially, it was using the coroutine scope, but after some changes, I was doing, it is not used and we don't really need it.
Let me update it.